src/plugins/orangehrmClaimPlugin/Api/ClaimExpenseAPI.php

```diff
@@ -0,0 +1,332 @@
+     *         description="Success",
+     *         @OA\JsonContent(
+     *             @OA\Property(
+     *                 property="data",
+     *                 type="array",
+     *                 @OA\Items(ref="#/components/schemas/Claim-ExpenseModel")
+     *             ),
+     *             @OA\Property(
+     *                 property="meta",
+     *                 type="object",
+     *                 @OA\Property(property="total", type="integer")
+     *             )
+     *         )
+     *     )
+     * )
+     * @inheritDoc
+     */
+    public function getAll(): EndpointResult
+    {
+        $claimExpenseSearchFilterParams = new ClaimExpenseSearchFilterParams();
+        $claimExpenseSearchFilterParams->setClaimRequestId($this->getRequestParams()->getIntOrNull(RequestParams::PARAM_TYPE_QUERY, self::PARAMETER_REQUEST_ID));
+        $claimExpenses = $this->getClaimService()->getClaimDao()->getClaimExpenseList($claimExpenseSearchFilterParams);
+        $count = $this->getClaimService()->getClaimDao()->getClaimExpenseCount($claimExpenseSearchFilterParams);
+        return new EndpointCollectionResult(ClaimExpenseModel::class, $claimExpenses, new ParameterBag([CommonParams::PARAMETER_TOTAL => $count]));
+    }
+
+    public function getValidationRuleForGetAll(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            $this->getValidationDecorator()->notRequiredParamRule(
+                new ParamRule(
+                    self::PARAMETER_REQUEST_ID,
+                    new Rule(Rules::POSITIVE)
+                )
+            ),
+        );
+    }
+
+    /**
+     * @OA\Post(
+     *     path="/api/v2/claim/expenses",
+     *     tags={"Claim/Expenses"},
+     *     @OA\RequestBody(
+     *         @OA\JsonContent(
+     *             type="object",
+     *             @OA\Property(property="expenseTypeId", type="integer"),
+     *             @OA\Property(property="amount", type="float"),
+     *             @OA\Property(property="requestId", type="integer"),
+     *             @OA\Property(property="note", type="string"),
+     *             @OA\Property(property="date", type="datetime"),
+     *             required={"name", "expenseTypeId", "amount", "requestId", "date"}
+     *         )
+     *     ),
+     *     @OA\Response(response="200",
+     *         description="Success",
+     *         @OA\JsonContent(
+     *             @OA\Property(
+     *                 property="data",
+     *                 ref="#/components/schemas/Claim-ExpenseModel"
+     *             ),
+     *             @OA\Property(property="meta", type="object")
+     *         )
+     *     )
+     * )
+     * @inheritDoc
+     */
+    public function create(): EndpointResult
+    {
+        $claimExpense = new ClaimExpense();
+        $claimExpense->getDecorator()->setClaimRequestByRequestId($this->getRequestParams()->getInt(RequestParams::PARAM_TYPE_BODY, self::PARAMETER_REQUEST_ID));
```

Comment:
can't put this inside **setClaimExpense()**

Reply:
since i use same method for both create and update. requestId can not be updated.